### ISSUE TITLE
Cannot Detect Extracted N64 Wii U Virtual Console Save Files (.sav)

### ISSUE THREAD

**Issue by PW5190** [Ninjiteu/N64SaveConverter#5]
When I tried to use this tool to convert my Wii U N64 saves to RetroArch and back to EverDrive 64, I noticed it cannot detect `.sav` files extracted from dumped Wii U Virtual Console saves using SaveMii on my softmodded console. Any chance this issue can be fixed?

**Comment by Ninjiteu** [comment]
I believe WiiU outputs eep files as .sav. So I'm 90% sure your .sav file is really an .eep file. Just change the extension from .sav to .eep and it should work fine. But just in case, be sure to use the lookup table in my app (click on the help tab at the top) and search for the game to check for the correct file type. (For example Super Mario 64 saves are EEP). And then change the extension to that file type and it should work.

**Comment by PW5190** [comment]
Also, I noticed that I cannot convert `.eep` saves from Wii/Wii U/ED64 to Project64/Mupen64. I can only convert those to a RetroArch `.srm` file first, then convert that `.srm` to a Project64/Mupen64 `.eep` file.

**Comment by PW5190** [comment]
> I believe WiiU outputs eep files as .sav.

Yes, and the same applies to fla and sra.

So, do the Wii and Wii U Virtual Console saves work on the EverDrive 64 without any conversion needed? (The first option listed shows Wii, Wii U, *and* EverDrive 64.)

**Comment by Ninjiteu** [comment]
Thats because you don't need to convert .eep from Wii/Wii U/ED64 to Project64/Mupen64. It should work as it is without any conversion. If it doesn't work, I recommend using my tool to pad to the standard file size. Just load your save into my app and click the pad to standard size checkbox. It will pad your file to 2 kilobytes. This should ensure the save file works for Project64/Mupen64.

**Comment by Ninjiteu** [comment]
Yes Wii and Wii U Virtual Console saves work on the EverDrive 64 without any conversion.

**Comment by PW5190** [comment]
Okay, I'll give importing my saves to the ED64 a try.

**Comment by Ninjiteu** [comment]
If the WiiU outputs all saves to .sav (including fla and sra), then you need to check the lookup table to make sure what the correct file type should be. What game are you trying to convert I can tell you right away if you have trouble with my lookup table.

**Comment by Ninjiteu** [comment]
fla and sra DO need to be converted when going from Wii/Wii U/ED64 to Project64/Mupen64 (and otherway). But eep and mpks do NOT need conversion when going from Wii/Wii U/ED64 to Project64/Mupen64 (and otherway around). Just to be clear

**Comment by PW5190** [comment]
I downloaded saves for three N64 games from a website called The Tech Game. One for Banjo-Kazooie, one for Pokémon Stadium, and one for Pokémon Stadium 2. The first one uses `.eep`, while the other two use `.fla`. Now I have a question. What source were these save files created on? These saves were uploaded to The Tech Game by a user named "Sean".

**Comment by Ninjiteu** [comment]
There's no way to know for sure where the files were created. The reason the source dropdown was included is because PC emulators (PJ64/Retroarch/Mupen) save the files in "little endian" while the consoles (Wii, WiiU, N64 Ed64) save the games in "big endian". This is important for fla and and sra saves because if you want to transfer saves between a little endian device to a big endian device, you need to byte swap the save file (which is what my tool does). For some reason eep and mpk saves don't need to be byteswapped (but on Daedalus for PSP and Vita they still need to be byteswapped. So I may end up creating a conversion option for eeps and mpks as well). Retroarch saves requires special padding and unpadding so thats why theres still a conversion option to go from PJ64 to retroarch for example (even though both are little endian).

Most likely those 3 saves are PC emulator saves. Put them in my tool and use the checkbox to pad them to the standard size and test them on PJ64 or Mupen64 (set the source to PJ64/Mupen64). I know PJ64 have special file names for the saves (you can't just name the save the same as the rom file. Start from the beginning and create an new save file. You must use the file name that appears there). I know for certain, the Banjo save should work anywhere you try it since its an eep (might need padding though but I doubt it). If either of the pokemon stadium saves don't work, feed it into my tool and convert them from wii/wiiu/ed64 fla to pj64/mupen64 fla. 

If you're still having trouble, you can email me your saves. I'll get them to work for you. danifalk@gmail.com
